### DISCUSSION TITLE
Can't seem to find VideoPlayer in the Bonsai.Vision.Design package

### DISCUSSION THREAD

**Discussion by LordKelv** [bonsai-rx/bonsai#1651] [Q&A] [answered]
Hello Bonsai-Rx team!

I visited your booth at SfN 2023 and was blown away by the ease and power of Bonsai.  

I've been trying to replicate a behavioral code I wrote in Matlab, but I stumped.  I need for a .avi video file to play after an input is sensed.  I went through your [documentation ](https://bonsai-rx.org/docs/api/Bonsai.Vision.Design.VideoPlayer.html#constructors) and `VideoPlayer`  should do it, but alas, I can't find `VideoPlayer` in the toolboxes.  I have reinstalled the [Bonsai.Vision.Design](https://bonsai-rx.org/docs/api/Bonsai.Vision.Design.html) package several times to see if it shows, but it doesn't. 

Any ideas on how I can get video playback after an input? 

Muchas gracias,
Kelvin

**Comment by bruno-f-cruz** [comment, marked as answer]
Do you mean this `FileCapture` perhaps?
![image](https://github.com/bonsai-rx/bonsai/assets/7049351/4e5ac39b-4823-4f4b-a5fb-ea903849e8ed)

**Comment by LordKelv** [comment]
That definitely plays the video, thanks!

On another note, I now have this:
![image](https://github.com/bonsai-rx/bonsai/assets/6320889/0f893d06-c66e-4d10-b78d-8a1ccdc8f5fa)

The video only plays if a key is held down.  How would I have the video play continuously after a single press?

**Reply by bruno-f-cruz** [reply]
You might want to check the documentation of SubscribeWhen operator

**Comment by LordKelv** [comment]
Oi, Bruno!
I've been trying this with the SubscribeWhen operator, but I'm still needing to hold down the key to have the video play in its entirety (not frame by frame as I press down any key):
![image](https://github.com/bonsai-rx/bonsai/assets/6320889/25134a08-4ea3-461d-a976-b2eeecab9cc8)
The value of the timer is 00:00:00 for the DueTime, and 00:00:05 for the Period.

**Comment by bruno-f-cruz** [comment]
Make KeyDown the trigger for the SubscribeWhen instead of the Timer.
Also, include the workflow file.

**Comment by LordKelv** [comment]
I swapped them, but still the video goes frame by frame.  I've attached the workflow as a zip file.
Thanks!
[play video when pressing button.zip](https://github.com/bonsai-rx/bonsai/files/13795494/play.video.when.pressing.button.zip)

**Reply by bruno-f-cruz** [reply]
Not sure what you mean by frame-by-frame, in your workflow, after clicking the `KeyDown`, the video will produce a frame every 5 seconds (period of the `Timer`). If you want to make it faster just decrease the period.
Alternatively, you can use the FileCapture node to take care of the playback by setting the property `Playing` to `True` and `PlaybackRate` to the rate you want the video to roughly play at.
![image](https://github.com/bonsai-rx/bonsai/assets/7049351/e29293ea-61cc-49ea-97d1-0a41dd96b382)
